config/src/main/java/org/springframework/security/config/web/server/ServerHttpSecurity.java

```diff
@@ -984,7 +986,7 @@ public ServerHttpSecurity oauth2Login(Customizer<OAuth2LoginSpec> oauth2LoginCus
 
 		private ServerWebExchangeMatcher authenticationMatcher;
 
-		private ServerAuthenticationSuccessHandler authenticationSuccessHandler = new RedirectServerAuthenticationSuccessHandler();
+		private ServerAuthenticationSuccessHandler authenticationSuccessHandler = null;
```

Comment:
No need for `= null` since it will default to `null`

Reply:
Fixed